<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Ivy MG &mdash; ReadTheDocs_Tutorial  documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=92fd9be5" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=19f00094" />

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js?v=5d32c60e"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="_static/documentation_options.js?v=5929fcd5"></script>
        <script src="_static/doctools.js?v=9a2dae69"></script>
        <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="#" class="icon icon-home">
            ReadTheDocs_Tutorial
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">Ivy MG</a></li>
<li><a class="reference internal" href="#getting-started">Getting Started</a><ul>
<li><a class="reference internal" href="#option-1-bioconda-installer">Option 1: Bioconda Installer</a></li>
</ul>
</li>
<li><a class="reference internal" href="#running-mercat2">Running Mercat2</a><ul>
<li><a class="reference internal" href="#dependencies">Dependencies</a></li>
<li><a class="reference internal" href="#external-dependencies">External Dependencies</a></li>
<li><a class="reference internal" href="#required-dependencies">Required dependencies</a></li>
<li><a class="reference internal" href="#notes-on-memory-usage-and-speed">Notes on memory usage and speed</a></li>
</ul>
</li>
<li><a class="reference internal" href="#options-for-using-mercat2">Options for using Mercat2</a><ul>
<li><a class="reference internal" href="#usage">Usage</a></li>
<li><a class="reference internal" href="#usage-examples">Usage Examples</a></li>
</ul>
</li>
<li><a class="reference internal" href="#outputs">Outputs</a><ul>
<li><a class="reference internal" href="#output-folders">Output Folders</a></li>
<li><a class="reference internal" href="#diversity-estimation">Diversity Estimation</a></li>
</ul>
</li>
<li><a class="reference internal" href="#citations-and-copywrite">Citations and Copywrite</a><ul>
<li><a class="reference internal" href="#citing-mercat2">Citing Mercat2</a></li>
<li><a class="reference internal" href="#copywrite">Copywrite</a></li>
</ul>
</li>
</ul>
</div>
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">ReadTheDocs_Tutorial</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="#" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Ivy MG</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <div class="toctree-wrapper compound">
</div>
<section id="ivy-mg">
<h1>Ivy MG<a class="headerlink" href="#ivy-mg" title="Link to this heading"></a></h1>
<p><a class="reference external" href="https://doi.org/10.1093/bioadv/vbae061"><img alt="paper" src="https://camo.githubusercontent.com/4e62fbf1240e11569ff5a5a55cfd46dd12aa4dfa41c1edd821f4f887d8e05c61/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f70617065722d42696f696e666f726d6174696373416476616e6365732d7465616c2e7376673f7374796c653d666c61742d737175617265266d61784167653d33363030" /></a> <a class="reference external" href="https://doi.org/10.1101/2022.11.22.517562"><img alt="preprint" src="https://camo.githubusercontent.com/4b1ce666560094f833328c23d008ff7aa59e483db5f38e2aa2c6fd9e58f133e7/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f7072657072696e742d42696f527869762d7265642e7376673f7374796c653d666c61742d737175617265266d61784167653d33363030" /></a> <a class="reference external" href="http://bioconda.github.io/recipes/mercat2/README.html"><img alt="install with" src="https://camo.githubusercontent.com/2b3da1f4ac43c9ff4aa3151c2ea27dd4d00f147e01dfb55225b08b1cbc284774/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f696e7374616c6c253230776974682d62696f636f6e64612d627269676874677265656e2e7376673f7374796c653d666c6174" /></a> <a class="reference external" href="https://anaconda.org/bioconda/mercat2"><img alt="Anaconda.org" src="https://camo.githubusercontent.com/8c1a83f3189c641f0f819a71dae7cabf3f7a002314f13621efba4d450a06d8a4/68747470733a2f2f616e61636f6e64612e6f72672f62696f636f6e64612f6d6572636174322f6261646765732f76657273696f6e2e737667" /></a> <a class="reference external" href="https://anaconda.org/bioconda/mercat2"><img alt="Last updated" src="https://camo.githubusercontent.com/ea29e40bfb15292c35d60919e24035ef541a309047f53a7f24b1b72798a5c0d4/68747470733a2f2f616e61636f6e64612e6f72672f62696f636f6e64612f6d6572636174322f6261646765732f6c61746573745f72656c656173655f646174652e737667" /></a> <a class="reference external" href="https://anaconda.org/bioconda/mercat2"><img alt="last updated" src="https://camo.githubusercontent.com/d1326f3d872ad8a6bb39ea6bebddc2d8440d8d918152f809326f33fa67941bf7/68747470733a2f2f616e61636f6e64612e6f72672f62696f636f6e64612f6d6572636174322f6261646765732f6c61746573745f72656c656173655f72656c61746976655f646174652e737667" /></a> <a class="reference external" href="https://anaconda.org/bioconda/mercat2"><img alt="Platforms" src="https://camo.githubusercontent.com/8741ef138a4ddda6adc435f3e84f64e608d4ca7d424f0fd62fd95bc240267de5/68747470733a2f2f616e61636f6e64612e6f72672f62696f636f6e64612f6d6572636174322f6261646765732f706c6174666f726d732e737667" /></a> <a class="reference external" href="https://anaconda.org/bioconda/mercat2"><img alt="license" src="https://camo.githubusercontent.com/ee552931b73896308a277e32be730377a4376610365ef0dc58a61f500426a5a6/68747470733a2f2f616e61636f6e64612e6f72672f62696f636f6e64612f6d6572636174322f6261646765732f6c6963656e73652e737667" /></a> <a class="reference external" href="https://anaconda.org/bioconda/mercat2"><img alt="Downloads" src="https://camo.githubusercontent.com/01b40ed4b155269d1cffae11c7b4a8c55403b452281e55fbec22b29fab76a20d/68747470733a2f2f616e61636f6e64612e6f72672f62696f636f6e64612f6d6572636174322f6261646765732f646f776e6c6f6164732e737667" /></a> <a class="reference external" href="https://pepy.tech/project/mercat2"><img alt="downloads" src="https://camo.githubusercontent.com/74e56c6c0f62d5906fa442b38250ce6ab586ef15cac8be93e872e6446ec5cdfd/68747470733a2f2f7374617469632e706570792e746563682f62616467652f6d657263617432" /></a> <a class="reference external" href="https://pepy.tech/project/mercat2"><img alt="downloads/month" src="https://camo.githubusercontent.com/e1152b598d7f78fe56f7e7c79aad8c9263bf78cba4b532018fd39652569129f4/68747470733a2f2f7374617469632e706570792e746563682f62616467652f6d6572636174322f6d6f6e7468" /></a> <a class="reference external" href="https://pepy.tech/project/mercat2"><img alt="downloads/week" src="https://camo.githubusercontent.com/67f98fd521ce96ea507aa2a195da3fe30f8850447500fc7976bd513a140c6adc/68747470733a2f2f7374617469632e706570792e746563682f62616467652f6d6572636174322f7765656b" /></a></p>
<a class="reference internal image-reference" href="https://github.com/raw-lab/mercat2/blob/master/MerCat2.jpg?raw=true"><img alt="https://github.com/raw-lab/mercat2/blob/master/MerCat2.jpg?raw=true" src="https://github.com/raw-lab/mercat2/blob/master/MerCat2.jpg?raw=true" style="width: 600px;" /></a>
</section>
<section id="getting-started">
<h1>Getting Started<a class="headerlink" href="#getting-started" title="Link to this heading"></a></h1>
<p>There are two different ways of installing mercat2: Bioconda Installer and Source Installer</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Installing Conda should be done first before running any of these options on the terminal, download Conda to install the latest verison to your system.</p>
</div>
<section id="option-1-bioconda-installer">
<h2>Option 1: Bioconda Installer<a class="headerlink" href="#option-1-bioconda-installer" title="Link to this heading"></a></h2>
<ol class="arabic simple">
<li><p>Install mamba using conda</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mamba</span> <span class="n">create</span> <span class="o">-</span><span class="n">n</span> <span class="n">mercat2</span> <span class="o">-</span><span class="n">c</span> <span class="n">conda</span><span class="o">-</span><span class="n">forge</span> <span class="o">-</span><span class="n">c</span> <span class="n">bioconda</span> <span class="n">mercat2</span>
<span class="n">conda</span> <span class="n">activate</span> <span class="n">mercat2v</span>
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">..</span> <span class="n">note</span><span class="p">::</span>
</pre></div>
</div>
<blockquote>
<div><p>Make sure you install mamba in your base conda environment. We have found that mamba is faster than conda for installing packages and creating environments. Using conda might fail to resolve dependencies.</p>
</div></blockquote>
<p>2. Install MerCat2
This step is done via Bioconda git reset –hard &lt;commit-hash&gt;</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mamba</span> <span class="n">create</span> <span class="o">-</span><span class="n">n</span> <span class="n">mercat2</span> <span class="o">-</span><span class="n">c</span> <span class="n">conda</span><span class="o">-</span><span class="n">forge</span> <span class="o">-</span><span class="n">c</span> <span class="n">bioconda</span> <span class="n">mercat2</span>
<span class="n">conda</span> <span class="n">activate</span> <span class="n">mercat2</span>
</pre></div>
</div>
<p>::
Option 2: Source Installer
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1. Clone mercat2 from github
2. Run install_mercat2.sh to install all required dependencies</p>
<p>This script creates a conda environment for you</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">raw</span><span class="o">-</span><span class="n">lab</span><span class="o">/</span><span class="n">mercat2</span><span class="o">.</span><span class="n">git</span>
<span class="n">cd</span> <span class="n">mercat2</span>
<span class="n">bash</span> <span class="n">install_mercat2</span><span class="o">.</span><span class="n">sh</span>
<span class="n">conda</span> <span class="n">activate</span> <span class="n">mercat2</span>
</pre></div>
</div>
</section>
</section>
<section id="running-mercat2">
<h1>Running Mercat2<a class="headerlink" href="#running-mercat2" title="Link to this heading"></a></h1>
<section id="dependencies">
<h2>Dependencies<a class="headerlink" href="#dependencies" title="Link to this heading"></a></h2>
<p>MerCat2 runs on python version 3.9 and higher.</p>
</section>
<section id="external-dependencies">
<h2>External Dependencies<a class="headerlink" href="#external-dependencies" title="Link to this heading"></a></h2>
<p>MerCat2 can run without external dependencies based on the options used.</p>
</section>
<section id="required-dependencies">
<h2>Required dependencies<a class="headerlink" href="#required-dependencies" title="Link to this heading"></a></h2>
<ul class="simple">
<li><dl class="simple">
<dt>When raw read .fastq is given</dt><dd><ul>
<li><p><a class="reference external" href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">Fastqc</a></p></li>
<li><p><a class="reference external" href="https://github.com/OpenGene/fastp">Fastp</a></p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>For bacteria/archaea rich samples (-prod option)</dt><dd><ul>
<li><p><a class="reference external" href="https://github.com/hyattpd/Prodigal">Prodigcal</a></p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>For eukaryote rich samples or general applications (-fgs option)</dt><dd><ul>
<li><p><a class="reference external" href="https://github.com/unipept/FragGeneScanRs">FragGeneScanRs</a></p></li>
</ul>
</dd>
</dl>
</li>
<li><p>Use this script for FragGeneScanRs</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">install</span> <span class="o">-</span><span class="n">c</span> <span class="n">bioconda</span> <span class="n">fastqc</span> <span class="n">fastp</span> <span class="n">prodigal</span>
</pre></div>
</div>
<p>These are available through Bioconda, except FragGeneScanRs, which is included in the MerCat2 distribution.</p>
</section>
<section id="notes-on-memory-usage-and-speed">
<h2>Notes on memory usage and speed<a class="headerlink" href="#notes-on-memory-usage-and-speed" title="Link to this heading"></a></h2>
<ul class="simple">
<li><p>MerCat2 uses a substantial amount of memory when the k-mer is high</p></li>
<li><p>Running MerCat2 on a personal computer using a k-mer length of ~4 should be OK</p></li>
<li><p>Total memory usage can be reduced using the Chunker feature (-s option), but keep in mind that in testing when the chunk size is too small (1MB) some of the least significant k-mers will get lost.</p></li>
<li><p>This does not seem to affect the overall results, but it is something to keep in mind. Using the chunker and reducing the number of CPUs available (-noption) can help reduce memory requirements.</p></li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The speed of MerCat2 can be increased when more memory or computer nodes are available on a cluster and using a chunk size of ~ 100Mb.</p>
</div>
</section>
</section>
<section id="options-for-using-mercat2">
<h1>Options for using Mercat2<a class="headerlink" href="#options-for-using-mercat2" title="Link to this heading"></a></h1>
<section id="usage">
<h2>Usage<a class="headerlink" href="#usage" title="Link to this heading"></a></h2>
<p>These are the different options that can be used with mercat2.py</p>
<p>usage: mercat2.py  [-h] [-i I [I …]] [-f F] -k K [-n N] [-c C] [-prod] [-fgs] [-s S] [-o O] [-replace] [-lowmem LOWMEM] [-skipclean] [-toupper] [-pca] [–version]</p>
<p>Example: mercat2.py -h</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 22%" />
<col style="width: 78%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Option</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>-h, –help</p></td>
<td><p>Shows this help message and exit</p></td>
</tr>
<tr class="row-odd"><td><p>–version, -v</p></td>
<td><p>Show the version number and exit</p></td>
</tr>
<tr class="row-even"><td><p>-i I [I …]</p></td>
<td><p>Path to input file(s)</p></td>
</tr>
<tr class="row-odd"><td><p>-f F</p></td>
<td><p>Path to folder containing input files</p></td>
</tr>
<tr class="row-even"><td><p>-k K</p></td>
<td><p>kmer length</p></td>
</tr>
<tr class="row-odd"><td><p>-n N</p></td>
<td><p>No of cores [auto detect]</p></td>
</tr>
<tr class="row-even"><td><p>-c C</p></td>
<td><p>Minimum kmer count [10]</p></td>
</tr>
<tr class="row-odd"><td><p>-prod</p></td>
<td><p>Run Prodigal on fasta files</p></td>
</tr>
<tr class="row-even"><td><p>-fgs</p></td>
<td><p>Run FragGeneScanRS on fasta files</p></td>
</tr>
<tr class="row-odd"><td><p>-s S</p></td>
<td><p>Split into x MB files. [100]</p></td>
</tr>
<tr class="row-even"><td><p>-o O</p></td>
<td><p>Output folder, default = ‘mercat_results’ in current directory</p></td>
</tr>
<tr class="row-odd"><td><p>-replace</p></td>
<td><p>Replace existing output directory [False]</p></td>
</tr>
<tr class="row-even"><td><p>-lowmem LOWMEM</p></td>
<td><p>Flag to use incremental PCA when low memory is available. [auto]</p></td>
</tr>
<tr class="row-odd"><td><p>-skipclean</p></td>
<td><p>Skip trimming of fastq files</p></td>
</tr>
<tr class="row-even"><td><p>-toupper</p></td>
<td><p>Convert all input sequences to uppercase</p></td>
</tr>
<tr class="row-odd"><td><p>-pca</p></td>
<td><p>Create interactive PCA plot of the samples (minimum of 4 fasta files required)</p></td>
</tr>
</tbody>
</table>
<p>Mercat2 assumes the input file format based on the extension provided</p>
<ul class="simple">
<li><p>Raw fastq file: [‘.fastq’, ‘.fq’]</p></li>
<li><p>Nucleotide fasta: [‘.fa’, ‘.fna’, ‘.ffn’, ‘.fasta’]</p></li>
<li><p>Amino acid fasta: [‘.faa’]</p></li>
<li><p>It also accepts gzipped versions of these filetypes with the added ‘.gz’ suffix</p></li>
</ul>
</section>
<section id="usage-examples">
<h2>Usage Examples<a class="headerlink" href="#usage-examples" title="Link to this heading"></a></h2>
<table class="docutils align-default">
<colgroup>
<col style="width: 22%" />
<col style="width: 78%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Type</p></th>
<th class="head"><p>Script</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Protein file (protein fasta - ‘.faa’)</p></td>
<td><p>mercat2.py -i file-name.faa -k 3 -c 10</p></td>
</tr>
<tr class="row-odd"><td><p>Nucleotide file (nucleotide fasta - ‘.fa’, ‘.fna’, ‘.ffn’, ‘.fasta’)</p></td>
<td><p>mercat2.py -i file-name.fna -k 3 -n 8 -c 10</p></td>
</tr>
<tr class="row-even"><td><p>Nucleotide file raw data (nucleotide fastq - ‘.fastq’)</p></td>
<td><p>mercat2.py -i file-name.fastq -k 3 -n 8 -c 10</p></td>
</tr>
<tr class="row-odd"><td><p>Many samples within a folder</p></td>
<td><p>mercat2.py -f /path/to/input-folder -k 3 -n 8 -c 10</p></td>
</tr>
<tr class="row-even"><td><p>Sample with prodigal option (raw reads or nucleotide contigs - ‘.fa’, ‘.fna’, ‘.ffn’, ‘.fasta’, ‘.fastq’)</p></td>
<td><p>mercat2.py -i /path/to/input-file -k 3 -n 8 -c 10 -prod</p></td>
</tr>
<tr class="row-odd"><td><p>Sample with FragGeneScanRS option (raw reads or nucleotide contigs - ‘.fa’, ‘.fna’, ‘.ffn’, ‘.fasta’, ‘.fastq’)</p></td>
<td><p>mercat2.py -i /path/to/input-file -k 3 -n 8 -c 10 -fgs</p></td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The prodigal and FragGeneScanRS options run the k-mer counter on both contigs and produced amino acids</p>
</div>
</section>
</section>
<section id="outputs">
<h1>Outputs<a class="headerlink" href="#outputs" title="Link to this heading"></a></h1>
<section id="output-folders">
<h2>Output Folders<a class="headerlink" href="#output-folders" title="Link to this heading"></a></h2>
<p>Results are stored in the output folder (default ‘mercat_results’ of the current working directory)</p>
<ul class="simple">
<li><dl class="simple">
<dt>The ‘report’ folder contains an html report with interactive plotly figures</dt><dd><ul>
<li><p>If at least 4 samples are provided a PCA plot will be included in the html report</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>The ‘tsv’ folder contains counts tables in tab separated format</dt><dd><ul>
<li><p>If protein files are given, or the -prod  option, a .tsv file is created for each sample containing k-mer count, pI, Molecular Weight, and Hydrophobicity metrics</p></li>
<li><p>If nucleotide files are given a .tsv file is created for each sample containing k-mer count and GC content</p></li>
</ul>
</dd>
</dl>
</li>
<li><p>If .fastq raw reads files are used, a ‘clean’ folder is created with the clean fasta file.</p></li>
<li><p>If the  -prod option is used, a ‘prodigal’ folder is created with the amino acid .faa and .gff files</p></li>
<li><p>If the  -fgs option is used, a ‘fgs’ folder is created with the amino acid .faa file</p></li>
</ul>
</section>
<section id="diversity-estimation">
<h2>Diversity Estimation<a class="headerlink" href="#diversity-estimation" title="Link to this heading"></a></h2>
<a class="reference internal image-reference" href="https://raw.githubusercontent.com/raw-lab/mercat2/master/doc/PCA.png"><img alt="https://raw.githubusercontent.com/raw-lab/mercat2/master/doc/PCA.png" src="https://raw.githubusercontent.com/raw-lab/mercat2/master/doc/PCA.png" style="width: 600px;" /></a>
<table class="docutils align-default">
<colgroup>
<col style="width: 50%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Alpha diversity metrics provided</p></th>
<th class="head"><p>Beta diversity metrics provided</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>shannon</p></td>
<td><p>euclidean</p></td>
</tr>
<tr class="row-odd"><td><p>simpson</p></td>
<td><p>cityblock</p></td>
</tr>
<tr class="row-even"><td><p>simpson_e</p></td>
<td><p>braycurtis</p></td>
</tr>
<tr class="row-odd"><td><p>goods_coverage</p></td>
<td><p>canberra</p></td>
</tr>
<tr class="row-even"><td><p>fisher_alpha</p></td>
<td><p>chebyshev</p></td>
</tr>
<tr class="row-odd"><td><p>dominance</p></td>
<td><p>correlation</p></td>
</tr>
<tr class="row-even"><td><p>chao1</p></td>
<td><p>cosine</p></td>
</tr>
<tr class="row-odd"><td><p>chao1_ci</p></td>
<td><p>dice</p></td>
</tr>
<tr class="row-even"><td><p>ace</p></td>
<td><p>hamming</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>jaccard</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>mahalanobis</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>manhattan (same as City Block in this case)</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>matching</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>minkowski</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>rogerstanimoto</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>russellrao</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>seuclidean</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>sokalmichener</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>sokalsneath</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>sqeuclidean</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>yule</p></td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="citations-and-copywrite">
<h1>Citations and Copywrite<a class="headerlink" href="#citations-and-copywrite" title="Link to this heading"></a></h1>
<section id="citing-mercat2">
<h2>Citing Mercat2<a class="headerlink" href="#citing-mercat2" title="Link to this heading"></a></h2>
<p>If you are publishing results obtained using MerCat2, please cite:</p>
<p>Figueroa JL*, Redinbo A*, Panyala A, Colby S, Friesen M, Tiemann L, White III RA. 2024.
MerCat2: a versatile k-mer counter and diversity estimator for database-independent property analysis obtained from omics data
Bioinformatics Advances, vbae061 <a class="reference external" href="https://doi.org/10.1093/bioadv/vbae061">Bioinformatics Advances</a></p>
<p>The star indicates* Co-first authors</p>
<p>BioRxiv pre-print
Figueroa JL, Panyala A, Colby S, Friesen M, Tiemann L, White III RA. 2022.
MerCat2: a versatile k-mer counter and diversity estimator for database-independent property analysis obtained from omics data.</p>
<p><a class="reference external" href="https://www.biorxiv.org/content/10.1101/2022.11.22.517562v1">bioRxiv</a></p>
</section>
<section id="copywrite">
<h2>Copywrite<a class="headerlink" href="#copywrite" title="Link to this heading"></a></h2>
<p>This is copyrighted by University of North Carolina at Charlotte, Jose L. Figueroa III, Andrew Redinbo, and Richard Allen White III. All rights reserved. DeGenPrime is a bioinformatic tool that can be distributed freely for academic use only. Please contact us for commerical use. The software is provided “as is” and the copyright owners or contributors are not liable for any direct, indirect, incidental, special, or consequential damages including but not limited to, procurement of goods or services, loss of use, data or profits arising in any way out of the use of this software.</p>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Author.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>